src/utils/getFenceAndImportDiffsFromGit.ts

```diff
@@ -0,0 +1,249 @@
+            const newPath = !fencePatch.newFile().id().iszero()
+                ? fencePatch.newFile().path()
+                : null;
+            const oldPath = !fencePatch.oldFile().id().iszero()
+                ? fencePatch.oldFile().path()
+                : null;
+
+            const newFenceContentPromise: Promise<Config> | Config = newPath
+                ? (async () => {
+                      const indexEntry = await index.getByPath(newPath);
+                      const newFenceBlob = await repo.getBlob(indexEntry.id);
+                      return loadConfigFromString(
+                          normalizePath(newPath),
+                          newFenceBlob.content().toString('utf-8')
+                      );
+                  })()
+                : emptyFence(normalizePath(oldPath));
+            const oldFenceContentPromise: Promise<Config> | Config = oldPath
+                ? (async () => {
+                      const oldFenceEntry = await compareTree.getEntry(oldPath);
+                      const oldFenceBlob = await oldFenceEntry.getBlob();
+                      return loadConfigFromString(
+                          normalizePath(oldPath),
+                          oldFenceBlob.content().toString('utf-8')
+                      );
+                  })()
+                : emptyFence(normalizePath(newPath));
+
+            const [newFence, oldFence] = await Promise.all([
+                newFenceContentPromise,
+                oldFenceContentPromise,
+            ]);
+
+            const fenceDiff = getFenceDiff(oldFence, newFence);
+            if (fenceDiff) {
+                fenceAndImportDiffs.fenceDiffs.set(
+                    normalizePath(fencePatch.newFile().path()),
+                    fenceDiff
+                );
+            }
+        })
+    );
+
+    const loadSourcePatchesPromise = Promise.all(
+        sourcePatches.map(async sourcePatch => {
+            // an oid of zero means the object is not present in the
+            // git db (e.g. it was deleted or did not yet exist)
+            const newPath = !sourcePatch.newFile().id().iszero()
+                ? sourcePatch.newFile().path()
+                : null;
+            const oldPath = !sourcePatch.oldFile().id().iszero()
+                ? sourcePatch.oldFile().path()
+                : null;
+
+            if (!newPath) {
+                // only check files that actually exist now
+                return;
+            }
+
+            const newSourceImportsPromise: Promise<Set<string>> | Set<string> = newPath
+                ? (async () => {
+                      const indexEntry = await index.getByPath(newPath);
+                      const newSourceBlob = await repo.getBlob(indexEntry.id);
+                      return getTsImportSetFromSourceString(
+                          newSourceBlob.content().toString('utf-8')
+                      );
+                  })()
+                : new Set();

```

Comment:
You should never hit this case since you already filtered out cases where `newPath` is null.